src/resdk/resources/data.py

```diff
@@ -353,7 +357,49 @@ def download(self, file_name=None, field_name=None, download_dir=None):
             "{}/{}".format(self.id, fname)
             for fname in self.files(file_name, field_name)
         ]
-        self.resolwe._download_files(files, download_dir)
+        file_names = [fname for fname in self.files(file_name, field_name)]
+
+        self.resolwe._download_files(files=files, download_dir=download_dir)
+
+        return file_names
+
+    def download_and_rename(
+        self,
+        custom_file_name: str,
+        field_name: Optional[str] = None,
+        file_name: Optional[str] = None,
+        download_dir: Optional[str] = None,
+    ):
+        """Download and rename a single file from data object."""
+
+        if download_dir is None:
+            download_dir = os.getcwd()
+
+        new_file_path = os.path.join(download_dir, custom_file_name)
+
+        if os.path.exists(new_file_path):
+            logging.warning(
+                f"File with path '{new_file_path}' already exists. Skipping download."
+            )
+            return
+
+        file_names = self.download(
```

Comment:
Could be called by constructing kwargs. @gregorjerse if current implementation is not okay, I can change it.